dot/sync/chain_sync.go

```diff
@@ -711,10 +711,20 @@ func handleReadyBlock(bd *types.BlockData, pendingBlocks DisjointBlockSet, ready
 	// if we're expecting headers, validate should ensure we have a header
 	if bd.Header == nil {
 		block := pendingBlocks.getBlock(bd.Hash)
-		bd.Header = block.header
+		if block != nil {
```

Comment:
I would suggest to `if block == nil` then log and return, this will remove the need of `else` branch